Documents/API.md

```diff
@@ -0,0 +1,139 @@
+# HMDA Platform API
+
+## Public HTTP Endpoints
+
+* `/`
+    * `GET` - Root endpoint, with information about the HMDA Platform service. Used for health checks
+    
+    Example response, with HTTP code 200:
+    
+    ```json
+    {
+      "status": "OK",
+      "service": "hmda-api",
+      "time": "2016-06-17T13:54:10.725Z",
+      "host": "localhost"
+    }
+    ```
+    
+    
+
+* `/institutions`
+    * `GET` - List of Financial Institutions
+    
+    Example response, with HTTP code 200: 
+    
+    ```json
+    {
+      "institutions": [
+        {
+          "id": "12345",
+          "name": "First Bank",
+          "status": "active"
+        },
+        {
+          "id": "123456",
+          "name": "Second Bank",
+          "status": "inactive"
+        }
+      ]
+    }
+    ```
+
+* `/institutions/<institution>`
+    * `GET` - Details for Financial Institution
+    
+    Example response, with HTTP code 200: 
+    
+    ```json
+    {
+      "institution": {
+      "id": "12345",
+      "name": "First Bank",
+      "status": "active"
+    },
+      "filings": [
+        {
+          "id": "2017",
+          "fid": "12345",
+          "status": "not-started"
+        },
+        {
+          "id": "2016",
+          "fid": "12345",
+          "status": "completed"
+        }
+      ]
+    } 
+    ```
+    
+    
+* `/institutions/<institution>/filings/<period>`
+    * `GET` - Details for a filing
+    
+    Example response, with HTTP code 200:
+    
+    ```json
+   {
+     "filing": {
+     "id": "2017",
+     "fid": "12345",
+     "status": "not-started"
+   },
+   "submissions": [
+     {
+       "id": 3,
+       "submissionStatus": "created"
+     },
+     {
+       "id": 2,
+       "submissionStatus": "created"
+     },
+     {
+       "id": 1,
+       "submissionStatus": "created"
+     }
+   ]
+   } 
+    ```
+    
+* `/institutions/<institution>/filings/<period>/submissions`
```

Comment:
It is an interesting UX question.  Which is more annoying: 

1. Having to go cancel a submission before you can upload a new one.
1. Accidentally overwriting a submission you weren't actually done with yet.

In either case, I don't think we're required to have the two-step process.  The "Begin Filing" page doesn't have to write to the API before displaying the page, right?  That page could wait to do the POST until the upload.  This has the benefit of not having written anything if the user decides to cancel out of that page.